### ISSUE TITLE
Enable torch.nn.gelu

### ISSUE THREAD

**Issue by inatatsu** [torch-spyre/torch-spyre#146]
### What feature would you like to be added?

Run [torch.nn.gelu](https://docs.pytorch.org/docs/stable/generated/torch.nn.GELU.html) on Spyre.

### Why is this feature needed?

This operator is used by important models.

**Comment by inatatsu** [comment]
A simple test case crashes during memory allocation in flex:
```python
import numpy
import torch
import torch.nn as nn
import random
import numpy as np

def get_op_callable():
        return torch.nn.functional.gelu

class OpModule(nn.Module):
        def __init__(self):
                super().__init__()
                op=get_op_callable()
                self.op = op
        def forward(self, *args):
                return self.op(*args)

def SimpleIterate(device=None):
        seed = 123
        random.seed(seed)
        np.random.seed(seed)
        torch.manual_seed(seed)
        net = OpModule().to(device)
        backend = "inductor"
        net_compile = net if device == "cpu" else torch.compile(net, backend=backend)
        var0 = torch.rand(torch.Size([10, 729, 4304])).to(device)
        inputs = [var0]
        output = net_compile(*inputs)
        if isinstance(output, tuple):
                output = output[0]
        return output.detach()

out_cpu = SimpleIterate(device="cpu")
out_sen = SimpleIterate(device="spyre").to("cpu")

atol=1e-5
rtol=1e-4
same = torch.allclose(out_cpu, out_sen, atol=atol, rtol=rtol)
print("Output match:", same)

if not same:
        print("Max diff:", (out_cpu - out_sen).abs().max().item())
```
```
[W1126 08:19:35.510322625 OperatorEntry.cpp:154] Warning: Warning only once for all operators,  other operators may also be overridden.
  Overriding a previously registered kernel for the same operator and the same dispatch key
  operator: aten::view(Tensor(a) self, SymInt[] size) -> Tensor(a)
    registered at /pytorch/build/aten/src/ATen/RegisterSchema.cpp:6
  dispatch key: PrivateUse1
  previous kernel: registered at /pytorch/aten/src/ATen/LegacyBatchingRegistrations.cpp:1079
       new kernel: registered at /home/inagaki/dt-inductor/torch-spyre/codegen/outputs/spyre_torch_ops.cpp:6213 (function operator())
malloc(): unaligned tcache chunk detected
Signal Received: 6 (Aborted)
Signal Received from pid=43492
*** BACKTRACE ***
/opt/ibm/spyre/senlib/lib/libsenlib-dd2.so(+0x4499de)[0x7fe3982329de]
/lib64/libc.so.6(+0x3ebf0)[0x7fe45765dbf0]
/lib64/libc.so.6(+0x8bedc)[0x7fe4576aaedc]
/lib64/libc.so.6(raise+0x16)[0x7fe45765db46]
/lib64/libc.so.6(abort+0xd3)[0x7fe457647833]
/lib64/libc.so.6(+0x29172)[0x7fe457648172]
/lib64/libc.so.6(+0x95f87)[0x7fe4576b4f87]
/lib64/libc.so.6(+0x99ffc)[0x7fe4576b8ffc]
/lib64/libstdc++.so.6(_Znwm+0x1c)[0x7fe454eefcbc]
/home/inagaki/dt-inductor/sentient/runtime/lib/libflex.so(_ZNSt15__new_allocatorISt13_Rb_tree_nodeINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEE8allocateEmPKv+0x5e)[0x7fe39a34b15a]
/home/inagaki/dt-inductor/sentient/runtime/lib/libflex.so(_ZNSt16allocator_traitsISaISt13_Rb_tree_nodeINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEEE8allocateERS8_m+0x28)[0x7fe39a34b0c0]
/home/inagaki/dt-inductor/sentient/runtime/lib/libflex.so(_ZNSt8_Rb_treeINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES5_St9_IdentityIS5_ESt4lessIS5_ESaIS5_EE11_M_get_nodeEv+0x25)[0x7fe39a34b02b]
/home/inagaki/dt-inductor/sentient/runtime/lib/libutil.so(_ZNSt8_Rb_treeINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES5_St9_IdentityIS5_ESt4lessIS5_ESaIS5_EE14_M_create_nodeIJRKS5_EEEPSt13_Rb_tree_nodeIS5_EDpOT_+0x1c)[0x7fe3998da528]
/home/inagaki/dt-inductor/sentient/runtime/lib/libutil.so(_ZNKSt8_Rb_treeINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES5_St9_IdentityIS5_ESt4lessIS5_ESaIS5_EE11_Alloc_nodeclIRKS5_EEPSt13_Rb_tree_nodeIS5_EOT_+0x2f)[0x7fe3998cfc17]
/home/inagaki/dt-inductor/sentient/runtime/lib/libutil.so(_ZNSt8_Rb_treeINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES5_St9_IdentityIS5_ESt4lessIS5_ESaIS5_EE10_M_insert_IRKS5_NSB_11_Alloc_nodeEEESt17_Rb_tree_iteratorIS5_EPSt18_Rb_tree_node_baseSJ_OT_RT0_+0x9d)[0x7fe399882359]
/home/inagaki/dt-inductor/sentient/runtime/lib/libutil.so(_ZNSt8_Rb_treeINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES5_St9_IdentityIS5_ESt4lessIS5_ESaIS5_EE17_M_insert_unique_IRKS5_NSB_11_Alloc_nodeEEESt17_Rb_tree_iteratorIS5_ESt23_Rb_tree_const_iteratorIS5_EOT_RT0_+0x7c)[0x7fe3998535b4]
/home/inagaki/dt-inductor/sentient/runtime/lib/libutil.so(_ZNSt8_Rb_treeINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES5_St9_IdentityIS5_ESt4lessIS5_ESaIS5_EE22_M_insert_range_uniqueIPKS5_EENSt9enable_ifIXsrSt7is_sameIS5_NSt15iterator_traitsIT_E10value_typeEE5valueEvE4typeESI_SI_+0x64)[0x7fe39983d34e]
/home/inagaki/dt-inductor/sentient/runtime/lib/libutil.so(_ZNSt3setINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt4lessIS5_ESaIS5_EEC1ESt16initializer_listIS5_ERKS7_RKS8_+0x8b)[0x7fe39982aa59]
/home/inagaki/dt-inductor/sentient/runtime/lib/libutil.so(+0x15e239)[0x7fe399818239]
/home/inagaki/dt-inductor/sentient/runtime/lib/libutil.so(+0x16c8d9)[0x7fe3998268d9]
/home/inagaki/dt-inductor/sentient/runtime/lib/libutil.so(_ZN9deeptools11ConvertDataEPvPKvRK20data_conversion_info+0xd1c)[0x7fe39981e86f]
/home/inagaki/dt-inductor/sentient/runtime/lib/libflex.so(_ZN4flex15FlexDataConvert7ComputeEPNS_18FlexComputeContextE+0xa8)[0x7fe39a50da96]
/home/inagaki/dt-inductor/sentient/runtime/lib/libflex.so(_ZN4flex13GraphExecutor7ExecuteEPNS_9FlexFrameERKSt6vectorIN6sendnn6TensorESaIS5_EERKS3_INS4_11ConstTensorESaISA_EEl+0x23c9)[0x7fe39a316f6b]
/home/inagaki/dt-inductor/sentient/runtime/lib/libflex.so(_ZN4flex16SuperNodeContext13OffloadPerNmbEPNS_6PerNmbE+0x121)[0x7fe39a6a1727]
/home/inagaki/dt-inductor/sentient/runtime/lib/libflex.so(_ZN4flex16SuperNodeContext24PrepareAndOffloadPerNmbsEPNS_6PerNmbE+0x53f)[0x7fe39a6a12a7]
/home/inagaki/dt-inductor/sentient/runtime/lib/libflex.so(+0xccd274)[0x7fe39a69d274]
/home/inagaki/dt-inductor/sentient/runtime/lib/libflex.so(+0xcd4a16)[0x7fe39a6a4a16]
/home/inagaki/dt-inductor/sentient/runtime/lib/libflex.so(+0xcd48d6)[0x7fe39a6a48d6]
/home/inagaki/dt-inductor/sentient/runtime/lib/libflex.so(+0xcd484a)[0x7fe39a6a484a]
/home/inagaki/dt-inductor/sentient/runtime/lib/libflex.so(+0xcd45ab)[0x7fe39a6a45ab]
/home/inagaki/dt-inductor/sentient/runtime/lib/libflex.so(_ZN4flex16SuperNodeContext19PrepareExecuteAsyncESt10shared_ptrINS_17SuperNodeWorkItemEE+0x1575)[0x7fe39a69e84b]
/home/inagaki/dt-inductor/sentient/runtime/lib/libflex.so(_ZN4flex16SuperNodeContext11LaunchAsyncERKSt6vectorIN6sendnn6TensorESaIS3_EERKS1_INS2_11ConstTensorESaIS8_EERKSt10shared_ptrINS2_21FrameContextInterfaceEEl+0x5db)[0x7fe39a6a3377]
/home/inagaki/dt-inductor/sentient/runtime/lib/libsendnn_runtime.so(_ZN6sendnn11GraphLoader4CopyERKSt6vectorINS_6TensorESaIS2_EERKS1_INS_11ConstTensorESaIS7_EEm+0x323)[0x7fe39787504b]
/home/inagaki/dt-inductor/torch-spyre/torch_spyre/_C.so(_ZN5spyre19copy_host_to_deviceERKN2at6TensorES3_+0x2cf)[0x7fe39ac8166f]
/home/inagaki/dt-inductor/torch-spyre/torch_spyre/_C.so(_ZN5spyre15spyre_copy_fromERKN2at6TensorES3_b+0x7cb)[0x7fe39ac82c6b]
/home/inagaki/dt-inductor/torch-spyre/torch_spyre/_C.so(_ZN3c104impl28wrap_kernel_functor_unboxed_INS0_6detail24WrapFunctionIntoFunctor_INS_26CompileTimeFunctionPointerIFN2at6TensorERKS6_S8_bEXadL_ZN5spyre15spyre_copy_fromES8_S8_bEEEES6_NS_4guts8typelist8typelistIJS8_S8_bEEEEES9_E4callEPNS_14OperatorKernelENS_14DispatchKeySetES8_S8_b+0x2b)[0x7fe39ac8f6fb]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(_ZN2at4_ops10_copy_from4callERKNS_6TensorES4_b+0x1a6)[0x7fe3a1d089f6]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(+0x1b06229)[0x7fe3a0f0b229]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(_ZN2at6native5copy_ERNS_6TensorERKS1_b+0x5c)[0x7fe3a0f0c91c]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(_ZN2at4_ops5copy_4callERNS_6TensorERKS2_b+0x195)[0x7fe3a1bd4d55]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(_ZN2at6native8_to_copyERKNS_6TensorESt8optionalIN3c1010ScalarTypeEES4_INS5_6LayoutEES4_INS5_6DeviceEES4_IbEbS4_INS5_12MemoryFormatEE+0x1735)[0x7fe3a11e4755]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(+0x2c0492d)[0x7fe3a200992d]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(_ZN2at4_ops8_to_copy10redispatchEN3c1014DispatchKeySetERKNS_6TensorESt8optionalINS2_10ScalarTypeEES7_INS2_6LayoutEES7_INS2_6DeviceEES7_IbEbS7_INS2_12MemoryFormatEE+0xf8)[0x7fe3a16a0a18]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(+0x29a639d)[0x7fe3a1dab39d]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(_ZN2at4_ops8_to_copy10redispatchEN3c1014DispatchKeySetERKNS_6TensorESt8optionalINS2_10ScalarTypeEES7_INS2_6LayoutEES7_INS2_6DeviceEES7_IbEbS7_INS2_12MemoryFormatEE+0xf8)[0x7fe3a16a0a18]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(+0x48d60ff)[0x7fe3a3cdb0ff]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(+0x48d6670)[0x7fe3a3cdb670]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(_ZN2at4_ops8_to_copy4callERKNS_6TensorESt8optionalIN3c1010ScalarTypeEES5_INS6_6LayoutEES5_INS6_6DeviceEES5_IbEbS5_INS6_12MemoryFormatEE+0x223)[0x7fe3a17488e3]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(_ZN2at6native2toERKNS_6TensorESt8optionalIN3c1010ScalarTypeEES4_INS5_6LayoutEES4_INS5_6DeviceEES4_IbEbbS4_INS5_12MemoryFormatEE+0x10c)[0x7fe3a11e185c]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(+0x2d0f233)[0x7fe3a2114233]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_cpu.so(_ZN2at4_ops15to_dtype_layout4callERKNS_6TensorESt8optionalIN3c1010ScalarTypeEES5_INS6_6LayoutEES5_INS6_6DeviceEES5_IbEbbS5_INS6_12MemoryFormatEE+0x238)[0x7fe3a18e0008]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_python.so(_ZNK2at6Tensor2toEN3c1013TensorOptionsEbbSt8optionalINS1_12MemoryFormatEE+0xed)[0x7fe3b3293f6d]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_python.so(+0x3dacc9)[0x7fe3b3238cc9]
/home/inagaki/dt-inductor/dti-venv/lib64/python3.12/site-packages/torch/lib/libtorch_python.so(+0x41d124)[0x7fe3b327b124]
/lib64/libpython3.12.so.1.0(+0x23bc10)[0x7fe457a62c10]
/lib64/libpython3.12.so.1.0(PyObject_Vectorcall+0x55)[0x7fe457a53d05]
/lib64/libpython3.12.so.1.0(_PyEval_EvalFrameDefault+0x95e)[0x7fe457a3889e]
/lib64/libpython3.12.so.1.0(PyEval_EvalCode+0xb0)[0x7fe457aff340]
/lib64/libpython3.12.so.1.0(+0x2fdcf8)[0x7fe457b24cf8]
/lib64/libpython3.12.so.1.0(+0x2f8dab)[0x7fe457b1fdab]
/lib64/libpython3.12.so.1.0(+0x3130a1)[0x7fe457b3a0a1]
/lib64/libpython3.12.so.1.0(_PyRun_SimpleFileObject+0x1c9)[0x7fe457b39979]
/lib64/libpython3.12.so.1.0(_PyRun_AnyFileObject+0x47)[0x7fe457b39087]
/lib64/libpython3.12.so.1.0(Py_RunMain+0x329)[0x7fe457b317d9]
/lib64/libpython3.12.so.1.0(Py_BytesMain+0x3d)[0x7fe457ae910d]
/lib64/libc.so.6(+0x295d0)[0x7fe4576485d0]
/lib64/libc.so.6(__libc_start_main+0x80)[0x7fe457648680]
python3(_start+0x25)[0x55c6eaed8095]
*****************
```

**Comment by inatatsu** [comment]
The crash above is due to that the input tensor is not padded. A padded input tensor results in `NotImplementedError`. I will look into it by assuming the padding is implemented.

**Comment by inatatsu** [comment]
We need to add a test case to this repo.
